docs/entity_creation.md

```diff
@@ -0,0 +1,207 @@
+
+- Querying to determine if the entity's properties are as expected and not malformed.
+
+#### Creating the handler trigger
+in `tests/e2e/entities/TestYourEntity.py`:
+
+Each test class relies upon a `SetUpClass` class method to ensure the prerequisite initialization and cleanup of the test environment. 
+
+The SetUp method of each Entity test class will look similar to the following:
+
+```
+0 class TestYourEntity(EntityTest):
+1     @classmethod
+2     def setUpClass(cls):
+3         super().setUpClass()
+4         cls.clean_db({"your_entity_table"})
+5 		  <trigger entity creation here>
+6 		  time.sleep(5)
+
+```
+
+
+- Starting on *L3* - ensuring that the environment for the test suite is initialised by calling the parent class' `SetUpClass` method - e.g. setting up psycopg & gql clients.
+- on *L4* - we truncate the tables, cleaning the database to ensure there are no rows left to interfere with our testing.
+- *L5* - Triggering the creation of our entity on this line
+- *L6* - a 5-second delay in order to allow the indexer time to capture our new entity
+
+#### SQL Database querying
+
+```
+0 def test_things(self):
+1   db_query = 'SELECT entity_field_a, entity_field_b, ... from test_entities'
+2 	entity = self.db_cursor.execute(self.db_query).fetchone()
+3 	self.assertIsNotNone(entity, "Assertion failure message")
+4 	self.assertEqual(entity[0], "correct value", "Assertion failure message") 
+```
+
+- L1 - Set out our SQL query to fetch the columns corresponding to the fields we defined in our schema.graphql.
+
+- L2 - Fetching the results of the query using the psycopg `db_cursor` database cursor object method [`execute`](https://www.psycopg.org/docs/usage.html) to return the first row.
+
+- L3 - firstly we assert that there is a row returned, as a non-null output of L2.
+
+- L3 - Assert the value of the first column returned of the row, e.g. "entity_field_a". Use the ordinal number of the column in the query as the index. In practise, this is abstracted within `tests/helpers/field_enums.py` with the use of an `enum`.
+
+#### GQL querying
+
+```	
+0   test_query = gql(
+1       """
+2       query getEntity {
+3           TestEntities {
+4               nodes {
+5                  entity_field_a
+6                  entity_field_b
+7                  entity_field_c
+8               }
+9           }
+10      }    
+11      """
+12   )
+13
+14   result = self.gql_client.execute(test_query)
+15   entity = result["TestEntities"]["nodes"]
+16   self.assertNotEqual(entity, [], "Assertion error message")
+17   self.assertEqual(entity[0]["entity_field_a"], "correct value", "Assertion error message")
+```
+
+- L0-12 - define the GraphQL query for the entity and relevant fields.
+- L14 - fetch the result of this query using the `gql_client` object method `execute`.
+- L16 - assert that the result is not null.
+- L17 - assert the actual field value against the expected value.
+
+
```

Comment:
We'll need to add a section about adding migrations. We can probably copy, move, and/or reference the [DB Migrations section of the README](https://github.com/fetchai/ledger-subquery#db-migrations) for most of this.